examples/gitlab/index.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express');
+const everyauth = require('@fusebit/everyauth-express');
+const { Gitlab } = require('@gitbeaker/node');
+const { v4: uuidv4 } = require('uuid');
+const cookieSession = require('cookie-session');
+
+const app = express();
+const port = 3000;
+
+// Get userId from the authorization redirect or via session if already authorized.
+const handleSession = (req, res, next) => {
+  if (req.query.userId) {
+    req.session.userId = req.query.userId;
+  }
+  if (!req.session.userId) {
+    return res.redirect('/');
+  }
+  return next();
+};
+
+app.use(
+  cookieSession({
+    name: 'session',
+    secret: 'secret',
+  })
+);
+
+app.set('view engine', 'pug');
+
+app.get('/', (req, res) => {
+  if (!req.session.userId) {
+    return res.redirect(`/authorize/${uuidv4()}`);
+  }
+  res.redirect('/finished');
+});
+
+app.use(
+  '/authorize/:userId',
+  (req, res, next) => {
+    if (!req.params.userId) {
+      return res.redirect('/');
+    }
```

Comment:
Will the url match w/o a userId?

Reply:
if a userId is not provided, it checks the session, if empty it initializes the authorization flow.

Reply:
No, I mean that `/authorize/:userId` won't match an inbound url `/authorize`, so the check doesn't make a lot of sense, unless I'm reading it wrong.

Reply:
you're right, doesn't make sense at all, no idea what  I was thinking with that middleware :P , removed!